transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/ForcedSender.java

```diff
@@ -0,0 +1,26 @@
+package com.google.android.datatransport.runtime;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.Discouraged;
+import androidx.annotation.WorkerThread;
+import com.google.android.datatransport.Priority;
+import com.google.android.datatransport.Transport;
+
+@Discouraged(
+    message =
+        "TransportRuntime is not a realtime delivery system, don't use unless you absolutely must.")
+public final class ForcedSender {
+  @WorkerThread
+  public static void sendBlockingWithPriority(Transport<?> transport, Priority priority) {
+    @SuppressLint("DiscouragedApi")
+    TransportContext context = getTransportContextOrThrow(transport).withPriority(priority);
+    TransportRuntime.getInstance().getUploader().upload(context, 1, () -> {});
```

Comment:
`upload()` is asynchronous, you want to use logAndUpdateState instead, see https://github.com/firebase/firebase-android-sdk/blob/fdb88d7c9a072b180bec441612a05337549a3d38/transport/transport-runtime-testing/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/UploadTestSupport.java#L28-L30